web_calendar_custom/static/src/js/web_calendar_custom.js

```diff
@@ -102,6 +102,32 @@ odoo.define('web_calendar_custom.renderer_setting', function (require) {
                         firstDay: String(config_values.calendar_first_day_of_week),
                         weekends: config_values.is_weekend_active,
                         slotEventOverlap: config_values.is_event_overlap,
+                        eventOverlap: function(stillEvent, movingEvent) {
+                            if(self.model == 'calendar.event'){
+                                if(movingEvent.record["start"]._d < Date.now() || movingEvent.record["state"] == "cancelled"){
+                                    return true;
+                                }
+                                if(typeof stillEvent.record["room_id"][0] != 'undefined' && typeof movingEvent.record["room_id"][0] != 'undefined'){
```

Comment:
It's good checking for undefined, but   What if I reserve only an instrument without the room (Possible scenario right? I tried it and there is no room_id on the event.... only intrument) The json dictionary will not have the key "room_id" and there will be an RerefenceError not undefined